Abs of real powers can be applied to the base

normalhuman opened the pull request:
#### References to other Issues or PRs

Fixes #14260

#### Brief description of what is fixed or changed
 
The current implementation of `Abs(Pow(base, exponent))` evaluates it only if the base is real. But when the exponent is real,, it is always true that we get `Pow(Abs(base), exponent)`. For integer powers this is true because Abs is multiplicative; this extends to rationals by considering roots, and then to all real numbers by density. Alternatively, consider that `x**y` is `exp(y*log(x))`, and taking absolute value amounts to taking the real part of the exponent. When y is real, it can be pulled out of `re(y*log(x))`.

For example, `Abs(I**n)` now evaluates to 1 when n is an integer symbol. In current master it returns
`sqrt(I**n*(-I)**n)` which isn't as easy to SymPy algorithms (in particular, `limit_seq`) to deal with.

The change is not applied when base contains symbols, because it would not result in evaluation (`Abs(x**n)` versus `Abs(x)**n`) and because some methods, like  `_simplifyconds` in integral transforms, expect symbolic expressions with Abs to satisfy `isinstance(expr, Abs)` -- [an example](https://github.com/sympy/sympy/blob/master/sympy/integrals/transforms.py#L909).

normalhuman committed:
Abs of integer powers can be applied to the base

The current implementation of Abs(Pow(base, exponent)) evaluates it only
if base is real. But when the exponent is an integer, it is always true
that we get Pow(Abs(base), exponent). For example, `Abs(I**n)` now
evaluates to 1 when n is an integer symbol. In current master it returns
`sqrt(I**n*(-I)**n)` which isn't as easy to SymPy algorithms (in
particular, `limit_seq`) to deal with.

normalhuman committed:
The same applies to real exponents, not only integer ones

Roland-Sands commented:
Would the following be of any use:

If `z` is a complex number, then write `z` in the form `r*exp(I*t)`, where r is a non-negative real number and t is a real number
So `abs(z) = r`
If `x` is a real number, then `z**x` is `(r**x)*exp(I*t*x)` and `abs(z**x)` is `r**x`

normalhuman committed:
Restrict the change to numeric base

The _simplifyconds method of sympy.integrals.transforms expects
expressions like Abs(x**2) to be an instance of Abs, so it breaks down
if it evaluates to Abs(x)**2
